### DISCUSSION TITLE
Removal of sp library from ufs-weather-model-env build with ip library version 5

### DISCUSSION THREAD

**Discussion by barlage** [JCSDA/spack-stack#1206] [Q&A] [answered]
I've recently installed the ufs-weather-model-env from spack-stack 1.7.0 on my MacBook (M3 Sonoma 14.5). Installation seems to have gone smoothly and I've used the environment to run other models using a small subset of modules. However, when attempting to compile ufs-weather-model, I get a dependency error for the sp library. This seems to be due to the removal of sp from the spack-stack build when ip@5 is used.

[`depends_on("sp", type="run", when="^ip@:4")`](https://github.com/JCSDA/spack-stack/blob/98bc12cc8e8ab626d1fad3f75f7df7721d9d2643/spack-ext/repos/spack-stack/packages/ufs-weather-model-env/package.py#L35)

When building ufs-weather-model, I assume due to this [line](https://github.com/ufs-community/ufs-weather-model/blob/73424e2d12af2632afbad7d26a1942df088423e8/CMakeLists.txt#L172), I get:

```
CMake Error at CMakeLists.txt:172 (find_package):
  By not providing "Findsp.cmake" in CMAKE_MODULE_PATH this project has asked
  CMake to find a package configuration file provided by "sp", but CMake did
  not find one.

  Could not find a package configuration file provided by "sp" (requested
  version 2.3.3) with any of the following names:

    spConfig.cmake
    sp-config.cmake

  Add the installation prefix of "sp" to CMAKE_PREFIX_PATH or set "sp_DIR" to
  a directory containing one of the above files.  If "sp" provides a separate
  development package or SDK, be sure it has been installed.

```

The current version of the ip library being built on my system using out of the box spack-stack is 5.0.0. Any thoughts on whether this is expected? Should I be pinning ip to a version 4 in this environment for ufs-weather-model builds as a solution?

**Comment by edwardhartnett** [comment, marked as answer]
THis is expected. The ip library, starting in the 5.0.0 release, includes all the sp functions. The sp library is deprecated and will no longer be supported. It is not needed if NCEPLIBS-ip-5.x is used.

So the sp dependency should be removed from the ufs-weather-model cmake file.

**Reply by barlage** [reply]
Thanks @edwardhartnett. Until ufs-weather-model updates this, I was able to build ufs-weather-model/develop using spack-stack 1.7.0 with ip@4.3.0
